internal/broker/broker_test.go

```diff
@@ -661,6 +669,21 @@ func TestEndSession(t *testing.T) {
 	require.NoError(t, err, "EndSession should not have returned an error when ending an existent session")
 }
 
+func TestUserPreCheck(t *testing.T) {
```

Comment:
I would really have a more comprehensive tests here.

Some with different brokers, like one not having the `ssh_allowed_suffixes` key, this is the main use case after all :)

Also, you can have one where the broker has more than one suffix as we only test one here.

So, I think if this, you end up with 5 test cases, which then justify table testing so that we don’t have to add it after we grew up in complexity as in other parts of authd where I needed to turn some tests to table later on :).